agent-manager-service/clients/openchoreosvc/utils.go

```diff
@@ -416,131 +424,295 @@ func toDeploymentDetailsResponse(sb *v1alpha1.ServiceBinding, environmentMap map
+				continue
+			}
+
+			// Extract image from spec.template.spec.containers[].image
+			if spec, ok := objMap["spec"].(map[string]interface{}); ok {
+				if template, ok := spec["template"].(map[string]interface{}); ok {
+					if podSpec, ok := template["spec"].(map[string]interface{}); ok {
+						if containers, ok := podSpec["containers"].([]interface{}); ok {
+							for _, container := range containers {
+								if containerMap, ok := container.(map[string]interface{}); ok {
+									// Look for the "main" container
+									if name, ok := containerMap["name"].(string); ok && name == "main" {
+										if image, ok := containerMap["image"].(string); ok {
+											return image
+										}
+									}
+								}
+							}
+						}
+					}
+				}
 			}
 		}
 	}
 
-	return steps
+	return ""
 }
 
-// calculateBuildPercentage determines completion percentage based on build conditions
-// Each step has specific percentage values: BuildInitiated=10%, BuildTriggered=40%, BuildCompleted=80%, WorkloadUpdated=100%
-func calculateBuildPercentage(conditions []metav1.Condition) *float32 {
-	percentage := float32(0)
+func MapConditionsToBuildSteps(conditions []metav1.Condition) []models.BuildStep {
+	steps := []models.BuildStep{
+		{Type: string(BuildStatusInitiated), Status: string(BuildStepStatusPending)},
+		{Type: string(BuildStatusTriggered), Status: string(BuildStepStatusPending)},
+		{Type: string(BuildStatusRunning), Status: string(BuildStepStatusPending)},
+		{Type: string(BuildStatusCompleted), Status: string(BuildStepStatusPending)},
+		{Type: string(WorkloadUpdated), Status: string(BuildStepStatusPending)},
+	}
 
-	// Check each condition and assign specific percentage values
-	for _, condition := range conditions {
-		if condition.Status == metav1.ConditionTrue {
-			switch condition.Type {
-			case string(ConditionBuildInitiated):
-				if percentage < 10 {
-					percentage = 10
-				}
-			case string(ConditionBuildTriggered):
-				if percentage < 40 {
-					percentage = 40
-				}
-			case string(ConditionBuildCompleted):
-				if percentage < 80 {
-					percentage = 80
-				}
-			case string(ConditionWorkloadUpdated):
-				if percentage < 100 {
-					percentage = 100
-				}
+	// Helper to find condition
+	findCondition := func(condType string) *metav1.Condition {
+		for i := range conditions {
+			if conditions[i].Type == condType {
+				return &conditions[i]
 			}
 		}
+		return nil
+	}
+
+	workflowCompleted := findCondition(string(ConditionWorkflowCompleted))
+	workflowRunning := findCondition(string(ConditionWorkflowRunning))
+	workflowSucceeded := findCondition(string(ConditionWorkflowSucceeded))
+	workflowFailed := findCondition(string(ConditionWorkflowFailed))
+	workloadUpdated := findCondition(string(ConditionWorkloadUpdated))
+
+	// Step 1: BuildInitiated (always succeeded if ComponentWorkflowRun exists)
+	steps[0].Status = string(BuildStepStatusSucceeded)
```

Comment:
Lets use constants instead of numbers for indices. Which would improve readability of the code.